### PR TITLE
fix: dsearch swap

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes mf-3276

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx [hunk 1]
```diff
@@ -161,22 +161,24 @@ export function TrendingView(props: TrendingViewProps) {
...
-                ? [
-                      {
-                          chainId: trending.coin.chainId,
-                          address: trending.coin.contract_address,
-                          pluginID: NetworkPluginID.PLUGIN_EVM,
-                      },
-                  ]
-                : []
+    const swapExpectedContract = useMemo(() => {
+        let contracts = trending?.contracts?.filter((x) => x.chainId && x.address)
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx]: const contracts = condition ? A : B

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx [hunk 2]
```diff
@@ -336,13 +338,13 @@ export function TrendingView(props: TrendingViewProps) {
...
-                                          isNativeTokenAddress(trending.coin.address)
+                                          swapExpectedContract.chainId ?? chainId,
+                                          isNativeTokenAddress(swapExpectedContract.address)
                                               ? SchemaType.Native
                                               : SchemaType.ERC20,
-                                          trending.coin.address ?? trending.coin.contract_address ?? '',
-                                          trending.coin.name,
-                                          trending.coin.symbol,
+                                          swapExpectedContract.address,
+                                          '',
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx:346]: Why delete this?

Comment (lelenei) [reply packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx:346]: The contract in the coin is not current chain

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx [hunk 3]
```diff
@@ -173,10 +165,20 @@ export function TrendingView(props: TrendingViewProps) {
...
                   ]
                 : []
-        const _contracts = contracts?.filter((x) => x.chainId === chainId) ?? []
-        if (_contracts.length > 0) return chainId
-        return contracts?.[0]?.chainId ?? chainId
-    }, [trending?.contracts, chainId])
+
+        const _contracts = (contracts.length ? contracts : fallbackContracts).filter((x) => x.chainId === chainId) ?? []
+        if (_contracts.length > 0) return _contracts[0]
+        return contracts?.[0]
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx]:
```suggestion
        return first(_contracts)
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx (hunk 2 above)
Comment (nuanyang233) [reply packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx:346]: Does this meet expectations?

Comment (lelenei) [reply packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingView.tsx:346]: yes

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**